libenkf/tests/enkf_workflow_job_test.c

```diff
@@ -39,34 +39,48 @@ ert_test_context_type * create_context( const char * config_file, const char * n
 void test_pre_simulation_copy__( ert_test_context_type * test_context,
                                  const char * job_name,
                                  const char * source_path,
-                                 const char * target_path,
-                                 const char * expected_DATA_ROOT) {
+                                 const char * target_path) {
 
   stringlist_type * args = stringlist_alloc_new();
-  unsetenv("DATA_ROOT");
   stringlist_append_copy( args , source_path );
-  stringlist_append_copy( args , target_path );
-  test_assert_true( ert_test_context_run_worklow_job( test_context , job_name , args) );
-
-  test_assert_string_equal( expected_DATA_ROOT , getenv("DATA_ROOT") );
-  if (expected_DATA_ROOT)
-    test_assert_true( util_is_directory( target_path ));
+  if (target_path)
+    stringlist_append_copy( args , target_path );
+  ert_test_context_run_worklow_job( test_context , job_name , args);
 
   stringlist_free( args );
 }
 
 void test_pre_simulation_copy(ert_test_context_type * test_context, const char * job_name , const char * job_file) {
+  enkf_main_type * enkf_main = ert_test_context_get_main( test_context );
+  model_config_type * model_config = enkf_main_get_model_config( enkf_main );
+  test_assert_false( model_config_data_root_set( model_config ));
+
   test_assert_true( ert_test_context_install_workflow_job( test_context , job_name , job_file ));
 
-  test_pre_simulation_copy__( test_context , job_name, "does_not_exist" , "target" , NULL );
+  test_pre_simulation_copy__( test_context , job_name, "does_not_exist" , "target");
 
   {
     FILE * f = util_mkdir_fopen("input/path/xxx/model/file", "w");
     fprintf(f, "File \n");
     fclose( f );
   }
-  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , "target" , "target");
-  test_assert_true( util_is_file( "target/model/file"));
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , NULL);
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , "target");
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model/file" , "target/extra_path");
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , "target/extra_path2");
+
+  model_config_set_data_root( model_config , "root");
+  test_assert_true( model_config_data_root_set( model_config ));
+
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , NULL );
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , "target");
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model/file" , "target/extra_path");
+  test_pre_simulation_copy__( test_context , job_name, "input/path/xxx/model" , "target/extra_path2");
+
+  test_assert_true( util_is_file( "root/model/file"));
```

Comment:
Just to be certain, please add the negated assert (checking that the file is not presented) before the copy is initialized.

Reply:
Length 1 is actually perfectly OK - but I have added log calls for the copy operation which is done.